salt/states/file.py

```diff
@@ -627,7 +627,7 @@ def _clean_dir(root, keep, exclude_pat):
             while True:
                 fn_ = os.path.dirname(fn_)
                 real_keep.add(fn_)
-                if fn_ in ['/', ''.join([os.path.splitdrive(fn_)[0], '\\\\'])]:
+                if fn_ in ['/', ''.join([os.path.splitdrive(fn_)[0], '\\'])]:
```

Comment:
Actually, in my testing, this is causing file.recurse to hang...
If `\\` fixes yours and `\\\\` fixes mine, then we need to do something like this:
```
                if fn_ in [
                        os.sep,
                        ''.join([os.path.splitdrive(fn_)[0], os.sep]),
                        ''.join([os.path.splitdrive(fn_)[0], os.sep, os.sep]),
                        ]:
```